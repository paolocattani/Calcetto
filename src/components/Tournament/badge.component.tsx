import React from 'react';
import { TournamentSelector } from '../../redux/selectors';
import { useSelector } from '../core/types';
import { Row, Col, Badge } from 'react-bootstrap';
import { formatDate } from '../../@common/utils/date.utils';
import { useTranslation } from 'react-i18next';

interface TournamentBadgeProps {}
const TournamentBadge: React.FC<TournamentBadgeProps> = () => {
	const tournament = useSelector(TournamentSelector.getTournament)!;
	const { t } = useTranslation(['tournament']);
	return (
		<Row>
			<Col>
				<h2 className="float-left">
					<Badge variant="info">
						<span>{t(`tournament:${tournament.public ? 'public' : 'private'}_tournament`)} </span>
						<strong>"{tournament.name.toUpperCase()}"</strong>
						<span>
							<i>
								{` - ${formatDate(tournament.date)} `}
								<small> @ {t(`tournament:progress.${tournament.progress}`)}</small>
								{process.env.NODE_ENV === 'development' ? <strong>{`  --> ( ${tournament.id} )`}</strong> : null}
							</i>
						</span>
					</Badge>
				</h2>
			</Col>
		</Row>
	);
};

export default TournamentBadge;
