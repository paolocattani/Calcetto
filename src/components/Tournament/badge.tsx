import React from 'react';
import { TournamentSelector } from 'redux/selectors';
import { useSelector } from 'react-redux';
import { Row, Col, Badge } from 'react-bootstrap';
import { formatDate } from 'components/core/utils';
import { useTranslation } from 'react-i18next';

interface TournamentBadgeProps {}
const TournamentBadge: React.FC<TournamentBadgeProps> = () => {
  const tournament = useSelector(TournamentSelector.getTournament)!;
  const { t } = useTranslation(['tournament']);
  return (
    <Row>
      <Col>
        <h2 className="float-left">
          <Badge variant="info">
            <span>{t('tournament:tournament')} </span>
            <strong>"{tournament.name.toUpperCase()}"</strong>
            <span>
              <i>
                {` - ${formatDate(tournament.date)} `}
                <small> @ {t(`tournament:progress.${tournament!.progress}`)}</small>
              </i>
            </span>
          </Badge>
        </h2>
      </Col>
    </Row>
  );
};

export default TournamentBadge;
