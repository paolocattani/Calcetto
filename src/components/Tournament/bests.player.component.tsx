import React, { useEffect, useState } from 'react';
import { Col } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import { PlayerDTO, StatsPairDTO, StatsPlayerDTO } from 'src/@common/dto';
import { StatsBestPairsResponse, StatsBestPlayersResponse } from 'src/@common/models';
import { fetchBestPairs, fetchBestPlayers } from 'src/redux/services/stats.service';

type StatsType = {
	pairs: { ever: Array<StatsPairDTO>; week: Array<StatsPairDTO> };
	players: { ever: Array<StatsPlayerDTO>; week: Array<StatsPlayerDTO> };
};
const aWeekAgo = new Date();
aWeekAgo.setDate(aWeekAgo.getDate() - 7);

interface TheBestsProps {}
// eslint-disable-next-line sonarjs/cognitive-complexity
const TheBests: React.FC<TheBestsProps> = () => {
	const MAX_RESULT = 3;
	const { t } = useTranslation(['stats']);
	const [loaded, setLoaded] = useState<boolean>(false);
	const [stats, setStats] = useState<StatsType>();
	useEffect(() => {
		if (!loaded) {
			(async () => {
				try {
					const { stats: bestPairsEver } = (await fetchBestPairs({})) as StatsBestPairsResponse;
					const { stats: bestPairsWeek } = (await fetchBestPairs({ from: aWeekAgo })) as StatsBestPairsResponse;
					const { stats: bestPlayerEver } = (await fetchBestPlayers({})) as StatsBestPlayersResponse;
					const { stats: bestPlayerWeek } = (await fetchBestPlayers({ from: aWeekAgo })) as StatsBestPlayersResponse;
					const model = {
						pairs: { ever: bestPairsEver, week: bestPairsWeek },
						players: { ever: bestPlayerEver, week: bestPlayerWeek },
					};
					console.log('useEffect : ', model);
					setStats(model);
				} catch (error) {
					console.log('Error');
				} finally {
					setLoaded(true);
				}
			})();
		}
	}, [loaded]);

	const getLabel = (player: PlayerDTO) => (player.alias ? player.alias : `${player.name} ${player.surname}`);
	const getPairs = (pairStats: Array<StatsPairDTO>, key: string) =>
		pairStats
			? pairStats
					.filter((v, i) => (pairStats.length >= MAX_RESULT ? i < MAX_RESULT : i > -1))
					.map((p) => (
						<p key={`${key}-${p.player1.id}`}>
							<code>
								{getLabel(p.player1)} - {getLabel(p.player2)}
							</code>
							<br />
						</p>
					))
			: 'not found';

	return (
		<>
			{stats && stats.pairs && (stats.pairs.ever || stats.pairs.week) ? (
				<article>
					<h2>{t('stats:best_pairs_ever')}</h2>
					<Col>{getPairs(stats.pairs.ever, 'ever')}</Col>
					<Col>{getPairs(stats.pairs.week, 'week')}</Col>
				</article>
			) : null}
		</>
	);
};

export default TheBests;
