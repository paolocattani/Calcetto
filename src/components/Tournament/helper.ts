import { getTodayDate } from '../core/utils';
import { TournamentModel } from './type';
import { useState } from 'react';

export function fetchTournaments(setterFunction: typeof useState, setterFunction2: typeof useState): void {
  (async () => {
    const response = await fetch('/api/tournament/list', {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' }
    });
    const result: TournamentModel[] = await response.json();
    let found = result.find(e => e.name === getTodayDate());
    let tmp = found ? [found, ...result] : result;
    tmp.sort();
    setterFunction(tmp);
    if (setterFunction2) setterFunction2(tmp[0]);
  })();
}

export function fetchTournament(setterFunction: typeof useState, tId: number): void {
  (async () => {
    const response = await fetch(`/api/tournament/${tId}`, {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' }
    });
    const result: TournamentModel = await response.json();
    setterFunction(result);
  })();
}

export async function getTournament(tId: number): Promise<TournamentModel | null> {
  const response = await fetch(`/api/tournament/${tId}`, {
    method: 'GET',
    headers: { 'Content-Type': 'application/json' }
  });
  const result: TournamentModel = await response.json();
  return response.ok ? result : null;
}
