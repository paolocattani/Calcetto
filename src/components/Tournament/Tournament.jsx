import React, { useState, useEffect } from 'react';
// React-Select
import Select from 'react-select';
// Bootstrap
import { Form, Button, Card, Col } from 'react-bootstrap';
import { useHistory } from 'react-router';
// Core
import { getTodayDate } from '../core/utils';
import { SessionContext, isEditable } from '../core/routing/SessionContext';
import { GenericToast } from '../core/generic/Commons';
// Helper
import { fetchTournaments, cardStyle, IndicatorSeparator } from './helper';
import { formatDate, translateTournamentProgress } from '../core/utils';
import NewTournament from './new';

const FTournament = () => {
  // State definition
  const [selectedOption, setSelectedOption] = useState(getTodayDate());
  const [tournamentList, setTournamentList] = useState([]);
  const [newTournament, setNewTournament] = useState(false);

  const messageInitialState = { message: '', type: 'success' };
  const [message, setMessage] = useState(messageInitialState);
  let currentHistory = useHistory();

  useEffect(() => fetchTournaments(setTournamentList, setSelectedOption), []);

  const showMessage = (message, type) => {
    setMessage({ message, type });
    setTimeout(() => setMessage(messageInitialState), 5000);
  };

  const handleSubmit = async event => {
    event.preventDefault();
    console.log('handleSubmit  : ', selectedOption);
    if (selectedOption) currentHistory.push(`/tournament/${selectedOption.id}`);
    else showMessage('Errore, riprovare piu tardi...', 'danger');
  };

  return (
    <>
      <GenericToast message={message.message} type={message.type} />
      <SessionContext.Consumer>
        {([session]) => (
          <Card style={cardStyle}>
            <Card.Header as="h2">Torneo</Card.Header>
            <Card.Body>
              <Col>
                {isEditable(session) && newTournament ? (
                  <NewTournament showMessage={showMessage} />
                ) : (
                  <Form onSubmit={handleSubmit}>
                    <label htmlFor="tournamentSelect">Selezione Torneo</label>
                    <Select
                      id="tournamentSelect"
                      components={{ IndicatorSeparator }}
                      styles={customStyles}
                      value={selectedOption}
                      options={tournamentList}
                      placeholder="Cerca un torneo"
                      isSearchable
                      getOptionLabel={getOptionLabel}
                      isClearable
                      onChange={selectedOption => setSelectedOption(selectedOption)}
                    />
                    <Button
                      type="submit"
                      size="md"
                      variant="outline-warning"
                      className="float-left default-color-white"
                      disabled={!selectedOption}
                    >
                      <span style={{ fontSize: 'larger', fontWeight: 'bolder' }}>Prosegui</span>
                    </Button>
                  </Form>
                )}
              </Col>
            </Card.Body>
            <Card.Footer>
              {isEditable(session) ? (
                newTournament ? (
                  <Button
                    type="button"
                    size="lg"
                    variant="outline-warning"
                    className="float-right default-color-white"
                    onClick={() => setNewTournament(false)}
                  >
                    Seleziona un torneo
                  </Button>
                ) : (
                  <Button
                    type="button"
                    size="lg"
                    variant="outline-warning"
                    className="float-right default-color-white"
                    onClick={() => setNewTournament(true)}
                  >
                    Crea un nuovo torneo
                  </Button>
                )
              ) : null}
            </Card.Footer>
          </Card>
        )}
      </SessionContext.Consumer>
    </>
  );
};

export default FTournament;

const getOptionLabel = ({ name, date, progress }) =>
  name + ' - ' + formatDate(date) + '@' + translateTournamentProgress(progress);

const customStyles = {
  // menuList: (provided, state) => ({ ...provided, border: '1px solid #ffc107' }),
  option: (provided, state) => ({
    ...provided,
    backgroundColor: 'white',
    color: 'black',
    '&:hover': {
      backgroundColor: '#64bd9c',
      color: 'white'
    }
  }),
  control: provided => ({ ...provided, height: '3vmin', marginBottom: '40px' }),
  singleValue: (provided, state) => ({ ...provided }),
  valueContainer: provided => ({ ...provided, height: '100%', fontSize: 'larger' })
};
