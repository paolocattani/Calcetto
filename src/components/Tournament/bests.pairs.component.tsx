import React, { useEffect, useState } from 'react';
import { Badge, Carousel, Col, Row } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import { StatsPairDTO } from 'src/@common/dto';
import { StatsBestPairsResponse } from 'src/@common/models';
import { formatDate, roundNumber } from 'src/@common/utils';
import { fetchBestPairs } from 'src/redux/services/stats.service';
import { LeftAngleIcon, LeftArrowIcon, RightAngleIcon, RightArrowIcon } from '../core/icons';
import { getLabel } from './helper';

type StatsType = {
	ever: Array<StatsPairDTO>;
	week: Array<StatsPairDTO>;
	month: Array<StatsPairDTO>;
};
const aWeekAgo = new Date();
aWeekAgo.setDate(aWeekAgo.getDate() - 7);
const aMonthAgo = new Date();
aMonthAgo.setMonth(aMonthAgo.getMonth() - 1);

interface TheBestsProps {}
// eslint-disable-next-line sonarjs/cognitive-complexity
const TheBests: React.FC<TheBestsProps> = () => {
	const MAX_RESULT = 4;
	const { t } = useTranslation(['stats']);
	const [loaded, setLoaded] = useState<boolean>(false);
	const [stats, setStats] = useState<StatsType>();
	const [index, setIndex] = useState(0);
	const handleSelect = (selectedIndex: number, event: Record<string, unknown> | null) => {
		setIndex(selectedIndex);
	};
	useEffect(() => {
		if (!loaded) {
			(async () => {
				try {
					const { stats: ever } = (await fetchBestPairs({})) as StatsBestPairsResponse;
					const { stats: week } = (await fetchBestPairs({ from: formatDate(aWeekAgo, '-') })) as StatsBestPairsResponse;
					const { stats: month } = (await fetchBestPairs({
						from: formatDate(aMonthAgo, '-'),
					})) as StatsBestPairsResponse;
					setStats({ ever, week, month });
				} catch (error) {
					console.log('Error');
				} finally {
					setLoaded(true);
				}
			})();
		}
	}, [loaded]);

	// https://www.w3schools.com/howto/howto_css_user_rating.asp
	const getPairs = (pairStats: Array<StatsPairDTO>, key: string) =>
		pairStats ? (
			<>
				<Row>
					<Col md={{ span: 2, offset: 2 }} style={{ textAlign: 'right' }}>
						{t('stats:pair')}
					</Col>
					<Col md="4" style={{ textAlign: 'left' }}>
						{t('stats:win_percentage')} ( {t(`stats:${key}`)} )
					</Col>
				</Row>
				{pairStats
					.filter((v, i) => (pairStats.length >= MAX_RESULT ? i < MAX_RESULT : i > -1))
					.sort(
						(p1, p2) =>
							roundNumber((p2.totwin / p2.totMatch) * 100, 1) - roundNumber((p1.totwin / p1.totMatch) * 100, 1)
					)
					.map((p, ii) => (
						<Row>
							<Col
								md={{ span: 2, offset: 2 }}
								style={{ textAlign: 'right', float: 'left', width: '15%', marginTop: '10px', fontSize: 'large' }}
							>
								<i>
									<strong>
										{getLabel(p.player1)} - {getLabel(p.player2)}
									</strong>
								</i>
							</Col>
							<Col md="4" style={{ marginTop: '10px', float: 'left', width: '70%' }}>
								<div style={{ width: '100%', backgroundColor: '#f1f1f1', textAlign: 'center', color: 'white' }}>
									<div
										style={{
											width: `${roundNumber((p.totwin / p.totMatch) * 100, 1)}%`,
											height: '18px',
											backgroundColor: '#ffc107',
										}}
									></div>
								</div>
							</Col>
							<Col
								md={{ span: 2 }}
								style={{ textAlign: 'left', float: 'left', width: '15%', marginTop: '10px', fontSize: 'large' }}
							>
								{roundNumber((p.totwin / p.totMatch) * 100, 1)}%
							</Col>
						</Row>
					))}
			</>
		) : null;

	return stats && Object.keys(stats).length > 0 ? (
		<Carousel
			touch
			activeIndex={index}
			onSelect={handleSelect}
			className="mt-3"
			prevIcon={<LeftAngleIcon size="3x" color="#343a40" />}
			nextIcon={<RightAngleIcon size="3x" color="#343a40" />}
		>
			{Object.keys(stats).map((key, ii) =>
				stats[key as keyof StatsType].length > 0 ? (
					<Carousel.Item style={{ opacity: 1 }} className="mb-5" key={`${key}-${ii * 18}`} interval={5000}>
						{getPairs(stats[key as keyof StatsType], key)}
					</Carousel.Item>
				) : null
			)}
		</Carousel>
	) : null;
};

export default TheBests;
