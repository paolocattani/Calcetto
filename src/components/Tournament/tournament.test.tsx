import React from 'react';
import Tournament from './wrapper.component';
import { render, RenderResult, fireEvent, screen } from '../../test/test-utils';
import userEvent from '@testing-library/user-event';
import { performAdminLogin, performUserLogin } from 'src/test/commons';
// import loginResponse from './_mocks_/login_response.json';

describe('<Tournament />.render', () => {
	// Admin
	describe('<Tournament />.render when user is type Admin', () => {
		let component: RenderResult;
		beforeEach(() => {
			const initialState = performAdminLogin();
			component = render(<Tournament />, { initialState });
		});
		it('should render correctly should match snapshot', () => {
			expect(component).toMatchSnapshot('Admin_tournament.tsx.snap');
		});
		it('should contains all elements', async () => {
			await screen.findByRole('button', { name: /continue/ });
			expect(screen.getByLabelText(/select/i)).not.toBeNull();
			// expect(screen.getByLabelText(/select/i).getAttribute('value')).toEqual('hFarm2 - 2020/09/05 @ progress.new');
			expect(screen.getByRole('button', { name: 'continue' })).not.toBeNull();
			expect(screen.queryByRole('button', { name: 'new' })).not.toBeNull();
		});
	});

	// User
	describe('<Tournament />.render when user is type User', () => {
		let component: RenderResult;
		beforeEach(() => {
			const initialState = performUserLogin();
			component = render(<Tournament />, { initialState });
		});
		it('should render correctly should match snapshot', () => {
			expect(component).toMatchSnapshot('User_tournament.tsx.snap');
		});
		it('should contains all elements', async () => {
			await screen.findByRole('button', { name: /continue/ });
			expect(screen.getByLabelText(/select/i)).not.toBeNull();
			// expect(screen.getByLabelText(/select/i).getAttribute('value')).toEqual('hFarm2 - 2020/09/05 @ progress.new');
			expect(screen.getByRole('button', { name: 'continue' })).not.toBeNull();
			expect(screen.queryByRole('button', { name: 'new' })).toBeNull();
		});
	});
});
