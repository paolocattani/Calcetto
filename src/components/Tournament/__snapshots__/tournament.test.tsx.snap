// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Tournament />.render <Tournament />.render when user is type Admin should render correctly should match snapshot: Admin_tournament.tsx.snap 1'] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="border-color: #ffc107; border-width: 3px; display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered"
        data-cy="loader"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
            style="color: whitesmoke; background-color: rgb(52, 58, 64); border-color: #ffc107;"
          >
            <div
              class="modal-title h4"
            >
              Caricamento....
            </div>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
              <span
                class="sr-only"
              >
                Close
              </span>
            </button>
          </div>
          <div
            class="text-center font-weight-bold modal-body"
            style="color: whitesmoke; background-color: rgb(52, 58, 64); border-color: #ffc107;"
          >
            Caricamento....
          </div>
          <div
            class="modal-footer"
            style="color: whitesmoke; background-color: rgb(52, 58, 64); border-color: #ffc107;"
          >
            <div
              class="spinner-border text-light"
            />
            <div
              class="spinner-border text-primary"
            />
            <div
              class="spinner-border text-secondary"
            />
            <div
              class="spinner-border text-success"
            />
            <div
              class="spinner-border text-danger"
            />
            <div
              class="spinner-border text-warning"
            />
            <div
              class="spinner-border text-info"
            />
            <div
              class="spinner-border text-dark"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports['<Tournament />.render <Tournament />.render when user is type User should render correctly should match snapshot: User_tournament.tsx.snap 1'] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
    style="padding-right: 0px;"
  >
    <div>
      <div
        class="row"
      >
        <div
          class="col-md-6 col-sm-12 offset-md-3"
        >
          <div
            class="card"
            data-cy="tournament-form"
            style="width: 100%; margin: auto; background-color: inherit; border-color: #ffc107; border-width: 3px; text-align: left;"
          >
            <h2
              class="card-header"
            >
              tournament
            </h2>
            <div
              class="card-body"
            >
              <div
                class="col"
              >
                <form
                  class=""
                >
                  <label
                    for="tournamentSelect"
                  >
                    select
                  </label>
                  <div
                    class=" css-2b097c-container"
                    id="tournamentSelect"
                  >
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-fjyw0r-control"
                    >
                      <div
                        class=" css-4pj5om-ValueContainer"
                      >
                        <div
                          class=" css-1uccc91-singleValue"
                        >
                          hFarm2 - 2020/09/05 @ progress.new
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class=""
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-label="select"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-3-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <div
                          aria-hidden="true"
                          class=" css-tlfecz-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                            />
                          </svg>
                        </div>
                        <span
                          style="align-self: stretch; background-color: green; margin-bottom: 8px; margin-top: 8px; margin-right: 10px; width: 1px;"
                        />
                        <div
                          aria-hidden="true"
                          class=" css-tlfecz-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                  <button
                    class="float-right default-color-white btn btn-outline-warning btn-lg"
                    data-cy="select-submit"
                    type="submit"
                  >
                    <span
                      style="font-size: larger; font-weight: bolder; padding: 1vw;"
                    >
                      continue
                    </span>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-arrow-alt-circle-right fa-w-16 fa-lg "
                      data-icon="arrow-alt-circle-right"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
                        fill="currentColor"
                      />
                    </svg>
                  </button>
                </form>
              </div>
            </div>
            <div
              class="card-footer"
            />
          </div>
        </div>
      </div>
      <div
        class="mt-2 row"
      >
        <div
          class="col"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="col-md-6 col-sm-12 offset-md-3"
      >
        <div
          class="card"
          data-cy="tournament-form"
          style="width: 100%; margin: auto; background-color: inherit; border-color: #ffc107; border-width: 3px; text-align: left;"
        >
          <h2
            class="card-header"
          >
            tournament
          </h2>
          <div
            class="card-body"
          >
            <div
              class="col"
            >
              <form
                class=""
              >
                <label
                  for="tournamentSelect"
                >
                  select
                </label>
                <div
                  class=" css-2b097c-container"
                  id="tournamentSelect"
                >
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    class=" css-fjyw0r-control"
                  >
                    <div
                      class=" css-4pj5om-ValueContainer"
                    >
                      <div
                        class=" css-1uccc91-singleValue"
                      >
                        hFarm2 - 2020/09/05 @ progress.new
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class=""
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-label="select"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-3-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                          />
                        </svg>
                      </div>
                      <span
                        style="align-self: stretch; background-color: green; margin-bottom: 8px; margin-top: 8px; margin-right: 10px; width: 1px;"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <button
                  class="float-right default-color-white btn btn-outline-warning btn-lg"
                  data-cy="select-submit"
                  type="submit"
                >
                  <span
                    style="font-size: larger; font-weight: bolder; padding: 1vw;"
                  >
                    continue
                  </span>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-arrow-alt-circle-right fa-w-16 fa-lg "
                    data-icon="arrow-alt-circle-right"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </form>
            </div>
          </div>
          <div
            class="card-footer"
          />
        </div>
      </div>
    </div>
    <div
      class="mt-2 row"
    >
      <div
        class="col"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
